core/src/main/java/software/amazon/awssdk/core/client/builder/SdkAsyncClientBuilder.java

```diff
@@ -42,27 +42,36 @@
     B asyncExecutorProvider(ExecutorProvider asyncExecutorProvider);
 
     /**
-     * Configures the HTTP client used by the service client. Either a client factory may be provided (in which case
-     * the SDK will merge any service specific configuration on top of customer supplied configuration) or provide an already
-     * constructed instance of {@link software.amazon.awssdk.http.async.SdkAsyncHttpClient}. Note that if an {@link
-     * software.amazon.awssdk.http.async.SdkAsyncHttpClient} is provided then it is up to the caller to close it when they are
-     * finished with it, the SDK will only close HTTP clients that it creates.
+     * Sets the {@link SdkAsyncHttpClient} that the SDK service client will use to make HTTP calls. This HTTP client may be
+     * shared between multiple SDK service clients to share a common connection pool. To create a client you must use an
+     * implementation specific builder. Note that this method is only recommended when you wish to share an HTTP client across
+     * multiple SDK service clients. If you do not wish to share HTTP clients, it is recommended to use
+     * {@link #asyncHttpClientBuilder(SdkAsyncHttpClient.Builder)} so that service specific default configuration may be applied.
+     *
+     * <p>Setting this will un-set any values set via {@link #asyncHttpClientBuilder(SdkAsyncHttpClient.Builder)}.</p>
```

Comment:
This may be surprising to have these two values step over each other. I think I'd rather just throw a runtime error if you've set both.

Reply:
Sure, will do.